tests/test_scalars.py

```diff
@@ -0,0 +1,663 @@
+    if TYPE_CHECKING_INVALID_USAGE:
+        md1 // td  # type: ignore[operator]
+        md2 // td  # type: ignore[operator]
+        md3 // td  # type: ignore[operator]
+        md4 // td  # type: ignore[operator]
+        md5 // td  # type: ignore[operator]
+        md6 // td  # type: ignore[operator]
+        md7 // td  # type: ignore[operator]
+        md8 // td  # type: ignore[operator]
+
+    check(assert_type(td / td, float), float)
+    check(assert_type(td / pd.NaT, float), float)
+    check(assert_type(td / md1, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td / md2, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td / md3, npt.NDArray[np.timedelta64]), np.ndarray)
+    check(assert_type(td / md4, npt.NDArray[np.timedelta64]), np.ndarray)
+    check(assert_type(td / md5, TimedeltaSeries), pd.Series)
+    check(assert_type(td / md6, TimedeltaSeries), pd.Series)
+    check(assert_type(td / md7, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td / md8, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    check(assert_type(pd.NaT / td, float), float)
+    # Note: None of the reverse truediv work
+    # TypeError: md1, md2, md3, md4, md5, md6, md7, md8
+    if TYPE_CHECKING_INVALID_USAGE:
+        md1 / td  # type: ignore[operator]
+        md2 / td  # type: ignore[operator]
+        md3 / td  # type: ignore[operator]
+        md4 / td  # type: ignore[operator]
+        # TODO: Series.__truediv__ says it supports Timedelta
+        #   it does not, in general, except for TimedeltaSeries
+        # md5 / td  # type: ignore[operator]
+        # md6 / td  # type: ignore[operator]
+        md7 / td  # type: ignore[operator]
+        md8 / td  # type: ignore[operator]
+
+
+def test_timedelta_mod_abs_unary() -> None:
+    td = pd.Timedelta("1 day")
+
+    with pytest.warns(FutureWarning):
+        i_idx = cast(pd.Int64Index, pd.Index([1, 2, 3], dtype=int))
+        f_idx = cast(pd.Float64Index, pd.Index([1.2, 2.2, 3.4], dtype=float))
+
+    check(assert_type(td % 3, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td % 3.5, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td % td, pd.Timedelta), pd.Timedelta)
+    check(
+        assert_type(td % np.array([1, 2, 3]), npt.NDArray[np.timedelta64]), np.ndarray
+    )
+    check(
+        assert_type(td % np.array([1.2, 2.2, 3.4]), npt.NDArray[np.timedelta64]),
+        np.ndarray,
+    )
+    int_series = pd.Series([1, 2, 3], dtype=int)
+    float_series = pd.Series([1.2, 2.2, 3.4], dtype=float)
+    check(assert_type(td % int_series, TimedeltaSeries), pd.Series)
+    check(assert_type(td % float_series, TimedeltaSeries), pd.Series)
+    check(assert_type(td % i_idx, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    check(
+        assert_type(td % f_idx, pd.TimedeltaIndex),
+        pd.TimedeltaIndex,
+    )
+
+    # mypy and pyright reports dt.timedelta, even though __abs__ returns Timedelta
+    check(assert_type(abs(td), dt.timedelta), pd.Timedelta)
```

Comment:
Definitely an issue with `pyright`, though not sure why it happens with `mypy`.  Reported for `pyright` here:  https://github.com/microsoft/pyright/issues/4082

Reply:
Thanks for reporting this.

Reply:
next version of `pyright` will have this fixed.

Reply:
With `pyright` fixing this, which also involved a change to typeshed, it may be that we are now getting `pd.Timedelta` back in typing, definitely for `pyright`, not sure about `mypy`.  If `mypy` fails, then do a `#type: ignore`

Reply:
Yes, had to ignore for mypy.